game-core/src/test/java/games/strategy/triplea/attachments/TechAbilityAttachmentTest.java

```diff
@@ -0,0 +1,103 @@
+package games.strategy.triplea.attachments;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import games.strategy.engine.data.GameParseException;
+
+public class TechAbilityAttachmentTest {
+
+  private final TechAbilityAttachment attachment = spy(new TechAbilityAttachment("", null, null));
+  private final String name = "Test Name";
+  private final String customToString = "CustomToString";
+
+  @BeforeEach
+  public void setup() {
+    when(attachment.toString()).thenReturn(customToString);
+  }
+
+  @Test
+  public void testSplitAndValidate_emptyString() {
+    final Exception e = assertThrows(GameParseException.class, () -> attachment.splitAndValidate(name, ""));
+    assertTrue(e.getMessage().contains(name));
+    assertTrue(e.getMessage().contains(customToString));
+  }
+
+  @Test
+  public void testSplitAndValidate_invalidLength() {
+    final Exception e = assertThrows(GameParseException.class, () -> attachment.splitAndValidate(name, "a:b:c"));
+    assertTrue(e.getMessage().contains(name));
+    assertTrue(e.getMessage().contains(customToString));
+  }
+
+  @Test
+  public void testSplitAndValidate_oneValue() throws Exception {
+    final String[] result = attachment.splitAndValidate(name, "a");
+    assertEquals("a", result[0]);
+    assertEquals(1, result.length);
```

Comment:
Nit: You probably want to assert the length expectation first.  Otherwise, the previous line will throw an `IndexOutOfBoundsException` effectively rendering this line useless because it will either always pass or never be reached.